sympy/functions/combinatorial/factorials.py

```diff
@@ -493,10 +494,21 @@ def _eval_is_positive(self):
             if n.is_odd:
                 return ((n + 1) / 2).is_even
 
-    def _eval_rewrite_as_gamma(self, n, piecewise=True, **kwargs):
-        from sympy import gamma, Piecewise, sqrt
-        return 2**(n/2)*gamma(n/2 + 1) * Piecewise((1, Eq(Mod(n, 2), 0)),
-                (sqrt(2/pi), Eq(Mod(n, 2), 1)))
+    def _eval_rewrite_as_cos(self, n, **kwargs):
+        from sympy import gamma, cos
+        return 2 ** (n / 2) * (2 / pi) ** ((1 - cos(pi*n))/4) * gamma(n/2 + 1)
+
+    def _eval_rewrite_as_gamma(self, n, piecewise=False, **kwargs):
+        from sympy import gamma, cos, Piecewise, sqrt
+
+        if piecewise:  # this seems wonky
```

Comment:
It is wonky. I don't think this parameter is needed.

Reply:
Thought so. I'll get rid of it.